### ISSUE TITLE
Don't warn "URIs must be properly escaped" in HTML5

### ISSUE THREAD

**Issue by jidanni** [htacg/tidy-html5#766]
Don't warn "URIs must be properly escaped" in HTML5.
See https://phabricator.wikimedia.org/T207963 .

**Comment by jidanni** [comment]
Yes, with 5.6.0
1. The user **should not have to use --fix-uri no**. Why?
because he has fed in **HTML5** where such URIs are **perfectly valid**.

He fed in HTML5, not HTML4.

2. (By the way, even if he does use --fix-uri no,
in the "Tidy found n warnings" n is reduced, but the
```
URIs must be properly escaped, they must not contain unescaped
```
message remains!)

**Comment by geoffmcl** [comment]
@jidanni yes, I now see this is a **bug** - sort of, in `HTML5`, valid utf-8 URIs are **valid**, and `libtidy` **fails** in this regard...

Quickly looking at the code in `void TY_(CheckUrl)( doc, node, attval)`, attempts have been made to accomodate `HTML5` **and** `HTML4` but is certainly **not** enough... thus agree the `fix-uri` option should **not** even apply in this case... there should be nothing found to **fix**... no URI error message output...

In one part it uses `static Bool IsURLCodePoint( ctmbstr p, uint *increment )`, but at the moment this part `looks` ok... but there are other **suspicious** parts of the service...

This `CheckUrl` service needs to be redone... re-thought for `html5`... reworked... enhanced, or replaced... fixed... etc...

For a start, at one point, it walks the URI text, **byte-by-byte**, and  bumps `escape_count` for each `(c > 0x7e)`, which is maybe ok for `HTML4`, but is totally **incorrect** for `HTML5`, and utf-8 text... so this sets up the `escape` message output problem... in valid utf-8 text there should be **nothing** to escape!

Have added [in_766.html](https://github.com/geoffmcl/tidy-test/blob/master/test/input5/in_766.html), a copy of the full file mentioned in  [T207963](https://phabricator.wikimedia.org/T207963), and [in_766-1.html](https://github.com/geoffmcl/tidy-test/blob/master/test/input5/in_766-1.html), a short version with more or less just the two URI marked **bad** by tidy, for consistent testing... hope this is ok...

It should be noted **both** these files **pass** W3C `nu` validation... and in this case, likewise should **pass** `libtidy`... at least for the URI...

In the full file, tidy also reports there are two empty `<ul>`, which can be removed with `--drop-empty-elements no`, but that is a separate issue... if it is an issue...

Look forward to comments, feedback, patches, PR, ... to address this utf-8 in html5 URI issue finally... thanks...

**Comment by jidanni** [comment]
I think you mean: --drop-empty-elements yes .

**Comment by jidanni** [comment]
https://github.com/jgm/pandoc/discussions/9905
